src/modules/parser.ts

```diff
@@ -221,14 +221,18 @@ function checkArgs (argsList: QueryType['operation']['args'], operationArg: stri
 
 /**
  * Parses the operation bit of the query
- * @param {queryType} query The JSON-Like query to be parsed
+ * @param {QueryType} query The JSON-Like query to be parsed
+ * @param {boolean} allowEmptyFields If 'true', empty fields are allowed (which is not considered a good practice)
  * @return {string} Parsed operation query
  */
-function parseOperation (query: QueryType): string {
+function parseOperation (query: QueryType, allowEmptyFields: boolean): string {
   let operation = query.operation
+  const hasEmptyFields = !operation.fields || !operation.fields.length
   info('Parsing operation %s', operation.name)
   if (!operation.name) throw new Error(`name is required for graphQL operation`)
-  if (!operation.fields) throw new Error(`field list is required for operation "${operation.name}"`)
+  if (hasEmptyFields && !allowEmptyFields) throw new Error(`field list is required for operation "${operation.name}"`)
+
+  if(hasEmptyFields) info('Hint: having no fields is not considered as a good practice')
```

Comment:
IMHO this hint is useful for unexperienced GraphQL devs, like me

Reply:
I agree!